src/components/history/History.tsx

```diff
@@ -0,0 +1,157 @@
+   */
+  useEffect(() => {
+    send({ type: "UPDATE_HISTORY_PROPS", props });
+  }, [props, send]);
+
+  /**
+   * Rebuild the file object whenever the history changes
+   */
+  const historyFile = useMemo(() => {
+    return new File(
+      [createHistoryFileContent(historyColumns, history)],
+      `${getTimestampStr()}_${historyProps.kind}_${historyProps.value}.tsv`,
+      {
+        type: "text/tsv",
+      }
+    );
+  }, [history, historyProps]);
+
+  /**
+   * Whenever the history file changes we need to rebuild the download URL
+   */
+  useEffect(() => {
+    const historyFileURL = URL.createObjectURL(historyFile);
+    setDownloadURL(historyFileURL);
+
+    /**
+     * Cleanup function that revokes the URL when it's no longer needed
+     */
+    return () => URL.revokeObjectURL(historyFileURL);
+  }, [historyFile, setDownloadURL]);
+
+  return (
+    <div data-testid="history">
+      {current.matches("error") && serverError && (
+        <Warning message={"History Search Error"} error={serverError}>
+          <WhiteButton onClick={() => send({ type: "RETRY" })}>
+            Retry
+          </WhiteButton>
+        </Warning>
+      )}
+
+      {current.matches("searching") && (
+        <div className="flex flex-row justify-center">
+          <LoadingSpinner />
+        </div>
+      )}
+
+      {current.matches("found") && history.length > 0 && (
+        <>
+          <div className="mt-6 mb-2 flex flex-row items-center justify-end space-x-3">
+            <p className="text-sm text-gray-700">
+              History for {historyDisplayValues[props.kind]}{" "}
+              <span className="font-medium">{props.value}</span>
+            </p>
+            <a href={downloadURL} download={true}>
+              <DownloadIcon name="Download" className="h-4 w-4 text-sdb" />
+            </a>
+          </div>
+          <DataTable columns={historyColumns} data={history} />
+        </>
+      )}
+    </div>
+  );
+}
+
+const historyColumns: Array<Column<HistoryTableEntry>> = [
+  {
+    Header: "Date",
+    accessor: "date",
+  },
+  {
+    Header: "Event Type",
+    accessor: "eventType",
+  },
+  {
+    Header: "Source",
+    accessor: "sourceBarcode",
+    Cell: (props: Cell<HistoryTableEntry>) => {
+      const barcode = props.row.original.sourceBarcode;
+      return <StyledLink to={`/labware/${barcode}`}>{barcode}</StyledLink>;
+    },
+  },
+  {
+    Header: "Destination",
+    accessor: "destinationBarcode",
+    Cell: (props: Cell<HistoryTableEntry>) => {
+      const barcode = props.row.original.destinationBarcode;
+      return <StyledLink to={`/labware/${barcode}`}>{barcode}</StyledLink>;
+    },
+  },
+  {
+    Header: "Sample ID",
+    accessor: "sampleID",
+  },
+  {
+    Header: "External ID",
+    accessor: "externalName",
+  },
+  {
+    Header: "Section Number",
+    accessor: "sectionNumber",
+  },
+  {
+    Header: "Biological State",
+    accessor: "biologicalState",
+  },
+  {
+    Header: "Process State",
```

Comment:
Can we call it "labware state" instead of "process state"? Otherwise "process state" will infect the vocabulary of the software, and I don't think it's very meaningful.

Reply:
Yep, good idea. Have made the change to labware state.